module02/linkedlist/MajorLinkedList.java

```diff
@@ -0,0 +1,131 @@
+package module02.linkedlist;
+
+public class MajorLinkedList<T> implements LinkedList<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MajorLinkedList() {
+        this.head = null;
+        this.size = 0;
+        this.tail = null;
+    }
+
+    @Override
+
+    public void addHead(T element) {
+        Node<T> newNode = new Node<>(element);
+        if (head == null) {
+            head = newNode;
+        } else {
+            Node<T> current = head;
+            while (current.next != null) {
+                current = current.next;
+            }
+            current.next = newNode;
+        }
+        size++;
+    }
+
+    @Override
+    public void remove(T element) {
+        if (head == null) return;
+        if (head.data.equals(element)) {
+            head = head.next;
+            size--;
+            return;
+        }
+        Node<T> current = head;
+        while (current.next != null) {
+            if (current.next.data.equals(element)) {
+                current.next = current.next.next;
+                size--;
+                return;
+            }
+            current = current.next;
+        }
+    }
+
+
+    @Override
+    public void clear() {
+        head = null;
+        size = 0;
+    }
+
+    @Override
+    public T getHead(int index) {
+        if (head == null) return null;
+        Node<T> current = head;
+        return current.data;
+    }
+
+    @Override
+    public void addTail(T element) {
+        if (tail == null) {
+            tail = new Node<>(element);
+        }
+    }
+
+    @Override
+    public void addIndex(T element, int index) {
+        Node<T> newNode = new Node<>(element);
+        if (head == null) {
+            head = newNode;
+        } else {
+            Node<T> current = head;
+            while (current.next != null) {
+                current = current.next;
+            }
+            current.next = newNode;
+        }
```

Comment:
The logic here is incorrect as it doesn't account for inserting at a specific index. It's just adding elements to the end of the list. You need to traverse the list up to the given index, then insert the new node there: